include/libdnf/module/module_query.hpp

```diff
@@ -133,6 +133,12 @@ class ModuleQuery : public libdnf::sack::Query<ModuleItem> {
     void filter_arch(
         const std::vector<std::string> & patterns, libdnf::sack::QueryCmp cmp_type = libdnf::sack::QueryCmp::EQ);
 
+    /// Group ModuleItems by `name`, `stream`, `context` and `arch`. Then within each group, keep packages that correspond with up to `limit` of latest `version`s in the group.
+    ///
+    /// @param limit            Keep `limit` number `version`s in each group.
+    /// @since 5.0.4
+    void filter_latest(int limit = 1);
```

Comment:
I see a problem in limit as int. If I understand the code correctly the code is unable to handle negative values. I checked PackageQueries and the latest filter also used int as limit argument but it supports also negative value.
I see some possibility how to resolve the issue

- document that negative value result in empty query.
  - plus add todo to support negative values
  - not support at all even not in future - add assertion - this is dangerous approach specially if the limit can be specified by user.
- not use int but unsigned int
  - difficult to extend functionality in future
  - different in comparison to PacageQueries

Reply:
Thank you! I totally forgot about the limit. I added the support for negative values right away, because I think it's quite straightforward. But if you disagree, I can change it back and add a todo.